### ISSUE TITLE
Plugins architecture

### ISSUE THREAD

**Issue by vvo** [algolia/algoliasearch-client-js#44]
## tl;dr;

There's no plugin infrastructure in the code, resulting in `if (angular)` `if (jquery)`.

Create one that allows us to remove the specific builds code from the main `algoliasearch.js`.

## TODO

- [ ] create a plugin way to create specific jQuery and Angular builds rather than storing the custom code in the main `algoliasearch.js` file.
- [ ] Remove specific builds code from main algoliasearch.js

## Boring stuff

We have [jQuery and Angular.js builds](https://github.com/algolia/algoliasearch-client-js/blob/4d28d22e7726b537c160190f337e2cafd68b0d63/README.md#jquery-angularjs) but the code needed by theses plugins is stored [inside the main algoliasearch.js](https://github.com/algolia/algoliasearch-client-js/blob/4d28d22e7726b537c160190f337e2cafd68b0d63/src/algoliasearch.js#L673) file.

This is not good because once we have 10 plugins we will be screwed and it complexifies the code a lot.

We must find a way to plug theses builds

We must have one entrypoint per plugin like today: [src/algoliasearch.angular.js](https://github.com/algolia/algoliasearch-client-js/blob/4d28d22e7726b537c160190f337e2cafd68b0d63/src/algoliasearch.angular.js) but it should store his own custom code.
